megamek/src/megamek/utilities/ai/ExtendedCostFunction.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2025 - The MegaMek Team. All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ *
+ */
+package megamek.utilities.ai;
+
+import java.util.*;
+
+/**
+ * Experimental set of extra parameters for the cost function. Can be "added" to any other cost function.
+ * @author Luana Coppio
+ */
+public class ExtendedCostFunction implements CostFunction {
+
+    private final CostFunction costFunction;
+
+    public ExtendedCostFunction(CostFunction costFunction) {
+        this.costFunction = costFunction;
+        StateVisitRegistry.reset();
+    }
+
+    @Override
+    public double resolve(UnitAction unitAction, Map<Integer, UnitState> currentUnitStates, Map<Integer, UnitState> nextUnitState, BehaviorParameters behaviorParameters) {
+        double baseCost = computeBaseCost(unitAction, currentUnitStates, behaviorParameters);
+
+        double boardControlTerm  = computeBoardControlTerm(currentUnitStates.get(unitAction.id()), nextUnitState);
+        double victoryTerm       = computeVictoryTerm(boardControlTerm);
+        double resourceTerm      = computeResourceTerm(currentUnitStates, unitAction, nextUnitState);
+        double curiosityTerm     = computeCuriosityTerm(nextUnitState.get(unitAction.id()));
+
+        // Weight for primary objective (victory)
+        double alpha = behaviorParameters.p23();
+        // Weight for resource-related term
+        double beta = behaviorParameters.p24();
+        // Weight for board-control term
+        double gamma = behaviorParameters.p25();
+        // Weight for exploration/curiosity term
+        double curiosityW = behaviorParameters.p26();
```

Comment:
I really think making these parameters named fields or something would be more maintainable going forward.